plugins/inputs/nsdp/nsdp.go

```diff
@@ -0,0 +1,119 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package nsdp
+
+import (
+	_ "embed"
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/tdrn-org/go-nsdp"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type NSDP struct {
+	Address     string          `toml:"address"`
+	DeviceLimit uint            `toml:"device_limit"`
+	Timeout     config.Duration `toml:"timeout"`
+
+	Log telegraf.Logger `toml:"-"`
+
+	conn *nsdp.Conn
+}
+
+func (*NSDP) SampleConfig() string {
+	return sampleConfig
+}
+
+func (n *NSDP) Init() error {
+	if n.Address == "" {
+		n.Address = nsdp.IPv4BroadcastTarget
+	}
+	if n.Timeout == 0 {
+		n.Timeout = config.Duration(2 * time.Second)
+	}
```

Comment:
See my comment in `init` (lowercase)
```suggestion

```

Reply:
See comment in init as well. I actually made this somewhat more restrictive to prevent non-working negative timeouts.

Reply:
See above with my updated suggestion.

Reply:
Adapted via 284fc600122546d38c8062a4f38bae035eb6dc93